XcodeServerSDK/XcodeServerConfig.swift

```diff
@@ -33,16 +42,51 @@ public class XcodeServerConfig : JSONSerializable {
         return dict
     }
     
-    public init(var host: String, user: String?, password: String?) {
+    ///
+    /// Initializes a server configuration with the provided host.
+    /// - parameter host: `Xcode` server host.
+    /// - paramater user: Username that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - parameter password: Password that will be used to authenticate against the `host` provided.
+    /// Can be `nil`.
+    ///
+    /// - returns: A fully initialized `XcodeServerConfig` instance.
+    ///
+    /// - throws:
+    ///     - `InvalidHostProvided`: When the host provided doesn't produce a valid `URL`
+    ///     - `InvalidSchemeProvided`: When the provided scheme is not `HTTPS`
+    ///
+    public required init(var host: String, user: String?, password: String?) throws {
+        guard let url = NSURL(string: host) else {
+            /*******************************************************************
```

Comment:
I think I have a couple of ideas for what you want to do; i'll try to play with it when I have the time.

Regarding the typo yeah; I noticed right after writing it but I guess Github doesn't send emails for updates on comments :joy: